lib/ignore/ignore.go

```diff
@@ -125,40 +127,42 @@ func New(fs fs.Filesystem, opts ...Option) *Matcher {
 }
 
 func (m *Matcher) Load(file string) error {
+	if m == nil {
+		return IsNil
+	}
```

Comment:
But Load() on a nil matcher is a clear error and should (imho) panic as it's always a bug.